### PR TITLE
Add mixin for gold linker

### PR DESCRIPTION
**Pull request by tylerjw**
I'm using this locally and it has proved useful.

### CODE REVIEW

**Comment by ruffsl** [comment]
@tylerjw , perhaps you could tack on or open another PR for adding a `lld.mixin` too?

File: gold.mixin [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "build": {
+        "gold": {
+            "cmake-args": [
+                "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=gold",
+                "-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=gold",
+                "-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=gold"
```
Comment (nuclearsandwich) [inline gold.mixin:7]:
```suggestion
                "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=gold",
                "-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=gold",
                "-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=gold",
                "-DCMAKE_STATIC_LINKER_FLAGS=-fuse-ld=gold"
```

Swapped the order of elements to be lexically sorted and added an entry for [CMAKE_STATIC_LINKER_FLAGS](https://cmake.org/cmake/help/v3.18/variable/CMAKE_STATIC_LINKER_FLAGS.html?highlight=linker_flags#variable:CMAKE_STATIC_LINKER_FLAGS).

Comment (nuclearsandwich) [reply gold.mixin:7]: Presuming the same feedback from https://github.com/colcon/colcon-mixin-repository/pull/28#discussion_r779054700 applies this suggestion is not correct.

Comment (nuclearsandwich) [inline gold.mixin:7]: I'd suggest lexically sorting JSON array elements unless the order is significant.

```suggestion
                "-DCMAKE_MODULE_LINKER_FLAGS=-fuse-ld=gold",
                "-DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=gold"
```

Comment (tylerjw) [reply gold.mixin:7]: It is not important afaik

**Review by nuclearsandwich — APPROVED** [review]

**Pull request merged**